src/solid.rs

```diff
@@ -13,6 +13,6 @@ pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
     } else {
         // ITRON error numbers are always negative, so we negate it so that it
         // falls in the dedicated OS error range (1..INTERNAL_START).
-        Err(NonZeroU32::new((-ret) as u32).unwrap().into())
```

Comment:
Ditto.

Reply:
`unwrap` here gets eliminated, but it still may be worth to use:
```rust
NonZeroU32::new(res.unsigned_abs()).unwrap_or(Error::UNEXPECTED)
```
to quickly prevent any concerns about potential panics.

Reply:
`Error::Unexpected` -> `Error::UNEXPECTED`.